src/sources/util/unix.rs

```diff
@@ -0,0 +1,71 @@
+use crate::event::Event;
+use crate::sources::Source;
+use bytes::Bytes;
+use futures::{future, sync::mpsc, Future, Sink, Stream};
+use std::{fs::canonicalize, path::PathBuf};
+use tokio::{
+    self,
+    codec::{FramedRead, LinesCodec},
+};
+use tokio_uds::UnixListener;
+use tracing::field;
+use tracing_futures::Instrument;
+
+/**
+* A function that takes a single line of a received message and builds an Event object.
+* Used by build_unix_source to encode any logic necessary for transforming the raw message
+* into a usuable Event object (for example decoding syslog messages).
+**/
+type BuildEventFn = fn(&str, Option<Bytes>, &str) -> Option<Event>;
```

Comment:
This should be doable with an `impl Fn`:

```
pub fn build_unix_source(...,
        build_event: impl Fn(&str, Option<Bytes>, &str) -> Option<Event>,
    ) -> Source
```

Alternately, but much more verbose, since the first parameter is constant to every call for a given use, make it a trait and store the `host_key` in the struct.

Reply:
Okay I took another crack at this and got it to work this time.  Adding the Clone trait and cloning the build_event closure before it was called in an inner move closure seemed to fix the borrowing/lifetime problems I was having.